### PR TITLE
Add infrastructure to enable 2-machine network testing on 1ES Azure pools. 

### PR DESCRIPTION
**Pull request by ProjectsByJackHe**
This PR adds the scripts and workflow templates to enable 2 machine testing on 1ES pools. 

Note: a lot of stuff can be re-factored and cleaned up for a better user experience. This should be done in a follow up PR, as this PR is already pretty massive. 

The work done here follows: 
image

### CODE REVIEW

File: .github/workflows/prepare-matrix.ps1 [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            $server | Add-Member -MemberType NoteProperty -Name "assigned_pool" -Value $Ubuntu2004Pool
+            $client | Add-Member -MemberType NoteProperty -Name "role" -Value "client"
+            $server | Add-Member -MemberType NoteProperty -Name "role" -Value "server"
+            $client | Add-Member -MemberType NoteProperty -Name "env_str" -Value $env_str
+            $server | Add-Member -MemberType NoteProperty -Name "env_str" -Value $env_str
+            $client | Add-Member -MemberType NoteProperty -Name "remote_powershell_supported" -Value 'FALSE'
+            $server | Add-Member -MemberType NoteProperty -Name "remote_powershell_supported" -Value 'FALSE'
+            $AzureJson += $client
+            $AzureJson += $server
+        }
```
Comment (nibanks) [inline .github/workflows/prepare-matrix.ps1:47]: Pull out the common commands and put them outside the `if` statements.

File: .github/workflows/prepare-matrix.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        required: true
+        type: string
+    outputs:
+      full-matrix:
+        description: "The full list of jobs to be run in secnetperf"
+        value: ${{ jobs.prepare-matrix.outputs.full-matrix }}
+      azure-matrix:
+        description: "The list of jobs to be run on Azure VMs"
+        value: ${{ jobs.prepare-matrix.outputs.azure-matrix }}
+permissions: write-all
```
Comment (nibanks) [inline .github/workflows/prepare-matrix.yml:35]: nit: add extra newline
```suggestion

permissions: write-all
```

File: .github/workflows/prepare-matrix.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Run Prepare Matrix
+      run: ./.github/workflows/prepare-matrix.ps1
+    - id: generate-full-matrix
+      run: echo "matrix=$(jq -c '.' ./.github/workflows/processed-matrix.json)" >> $GITHUB_OUTPUT
+      shell: bash
+    - id: generate-azure-matrix
+      run: echo "matrix=$(jq -c '.' ./.github/workflows/azure-matrix.json)" >> $GITHUB_OUTPUT
+      shell: bash
```
Comment (nibanks) [inline .github/workflows/prepare-matrix.yml:54]: Any reason these steps aren't part of the powershell itself?

Comment (nibanks) [reply .github/workflows/prepare-matrix.yml:54]: Talked on a call. Leaving as is.

File: .github/workflows/quic.yml [hunk 4]
```diff
@@ -139,35 +139,356 @@ jobs:
...
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      with:
+        repository: microsoft/msquic
+        ref: jackhe/1es-azure-pools
+    - name: Checkout netperf
+      if: ${{ matrix.role == 'server' && matrix.remote_powershell_supported == 'FALSE'}}
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      with:
+        repository: microsoft/netperf
+        ref: 'jackhe/1es-azure-pools'
```
Comment (nibanks) [inline .github/workflows/quic.yml:222]: remove?

**Review by nibanks — APPROVED** [review]

File: .github/workflows/quic.yml (hunk 4 above)
Comment (ProjectsByJackHe) [reply .github/workflows/quic.yml:222]: Done.

File: .github/workflows/prepare-matrix.ps1 (hunk 1 above)
Comment (ProjectsByJackHe) [reply .github/workflows/prepare-matrix.ps1:47]: Done.

**Pull request merged**